src/lsm/k_way_merge_benchmark.zig

```diff
@@ -0,0 +1,174 @@
+const std = @import("std");
+const assert = std.debug.assert;
+
+const stdx = @import("stdx");
+
+const Bench = @import("../testing/bench.zig");
+const Pending = error{Pending};
+const KWayMergeIteratorType = @import("k_way_merge.zig").KWayMergeIteratorType;
+
+const streams_count_max = 32;
+const repetitions: usize = 32;
+
+const body_fmt = "K={:_>2} L={:_>3} WT={}";
+
+// Those Values are close to the real-world use case.
+const Values = .{
+    ValueType(u64, 128),
+    ValueType(u256, 32),
+    ValueType(u256, 32),
+    ValueType(u256, 32),
+    ValueType(u128, 16),
+    ValueType(u64, 8),
+    ValueType(u128, 16),
+};
+
+test "benchmark: k-way-merge" {
+    var bench: Bench = .init();
+    defer bench.deinit();
+
+    bench.report("WT: Wall time/merged element", .{});
+
+    const streams_count: usize = @intCast(bench.parameter("streams_count", 4, 32));
+    const stream_length: usize = @intCast(bench.parameter("stream_length", 128, 8192));
+    assert(streams_count <= streams_count_max);
+
+    var prng = stdx.PRNG.from_seed(bench.seed);
+
+    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
+    defer arena.deinit();
+
+    const allocator = arena.allocator();
```

Comment:
Don't use `const allocator`, use `const arena` or `const gpa`, depending on whether the callee must free stuff or not. We probably want `arena` for benchmarks. 

It's an open question what to call the non-type-erased thing. 

In main.zig, we call it

```zig
    var arena_instance = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    defer arena_instance.deinit();

    // Arena is an implementation detail, all memory must be freed.
    const gpa = arena_instance.allocator();
```

(again, note how gpa vs arena explains you the contract)